core/src/banking_stage/consumer.rs

```diff
@@ -413,12 +413,20 @@ impl Consumer {
         let pre_results = vec![Ok(()); txs.len()];
         let check_results =
             bank.check_transactions(txs, &pre_results, MAX_PROCESSING_AGE, &mut error_counters);
-        let check_results = check_results.into_iter().map(|result| result.map(|_| ()));
+        // If checks passed, verify pre-compiles and continue processing on success.
+        let check_results: Vec<_> = txs
+            .iter()
+            .zip(check_results)
+            .map(|(tx, result)| match result {
+                Ok(_) => tx.verify_precompiles(&bank.feature_set),
+                Err(err) => Err(err),
+            })
+            .collect();
```

Comment:
Does this need to add another allocation?

Reply:
Discussed offline that this is to ensure that downstream timing isn't affected by this extra verification

Reply:
ok, probably deserves a comment at least